Dockerfile

```diff
@@ -1,45 +1,60 @@
 FROM ruby:2.3
 
-RUN apt-get update && \
-    apt-get install --no-install-recommends -y cron && \
-    rm -rf /var/lib/apt/lists/* && \
-    apt-get clean
+RUN SUPERCRONIC_URL=https://github.com/aptible/supercronic/releases/download/v0.1.3/supercronic-linux-amd64 && \
+    SUPERCRONIC_BIN=/usr/local/bin/supercronic && \
+    SUPERCRONIC_SHA1SUM=96960ba3207756bb01e6892c978264e5362e117e && \
+    curl -fsSL -o "${SUPERCRONIC_BIN}" "${SUPERCRONIC_URL}" \
+    && echo "${SUPERCRONIC_SHA1SUM}  ${SUPERCRONIC_BIN}" | sha1sum -c - \
+    && chmod +x "${SUPERCRONIC_BIN}"
 
 ENV RAILS_ENV=production \
+    RAILS_LOG_TO_STDOUT=true \
     RAILS_SERVE_STATIC_FILES=true
 
 WORKDIR /usr/src/app
-COPY . ./
-RUN echo $SOURCE_COMMIT > REVISION
 
-RUN buildDeps='libmagic-dev' && \
+# separately from rest of the code to avoid re-running bundler when Gemfile doesn't change
+# plugins is copied completely, because COPY can't pick wildcard directories
+COPY plugins plugins/
+COPY Gemfile Gemfile.lock ./
+
+RUN BUILD_DEPS='libmagic-dev' && \
     apt-get update && \
-    apt-get install --no-install-recommends -y $buildDeps && \
-    rm -rf /var/lib/apt/lists/* && \
-    bundle install --deployment --without development test && \
-    apt-get purge -y --auto-remove $buildDeps && \
-    echo "Foodsoft::Application.config.secret_token = ENV['SECRET_KEY_BASE']" > config/initializers/secret_token.rb && \
-    mkdir -p log && \
-    ln -sfn /dev/stdout log/production.log && \
-    bundle exec whenever --update-crontab
-
-# Add a temporary mysql-server for assets precompilation
+    apt-get install --no-install-recommends -y $BUILD_DEPS && \
+    echo 'gem: --no-document' >> ~/.gemrc && \
+    bundle config --global build.nokogiri "--use-system-libraries" && \
+    bundle install --deployment --without development test -j 4 && \
+    apt-get purge -y --auto-remove $BUILD_DEPS && \
+    rm -Rf /var/lib/apt/lists/* /var/cache/apt/*
+
+COPY . ./
+
+RUN mv config/app_config.yml.SAMPLE config/app_config.yml
+
+# compile assets with temporary mysql server
 RUN export DATABASE_URL=mysql2://localhost/temp && \
+    export SECRET_KEY_BASE=thisisnotimportantnow && \
     export DEBIAN_FRONTEND=noninteractive && \
     apt-get update && \
     apt-get install -y mysql-server && \
     /etc/init.d/mysql start && \
-    cp config/app_config.yml.SAMPLE config/app_config.yml && \
-    bundle exec rake db:setup && \
-    bundle exec rake assets:precompile && \
-    rm config/app_config.yml && \
-    rm -rf tmp/* && \
+    bundle exec rake db:setup assets:precompile && \
+    rm -Rf tmp/* && \
     /etc/init.d/mysql stop && \
-    rm -rf /run/mysqld /tmp/* /var/lib/mysql /var/log/mysql* && \
+    rm -Rf /run/mysqld /tmp/* /var/lib/mysql /var/log/mysql* && \
     apt-get purge -y --auto-remove mysql-server && \
-    rm -rf /var/lib/apt/lists/*
+    rm -Rf /var/lib/apt/lists/* /var/cache/apt/*
+
+RUN bundle exec whenever >crontab
```

Comment:
I'd keep the call to whenever int the previous `RUN`

Reply:
Logically it's a different topic, isn't this clearer when reading?
Comments within a single `RUN` don't really work, do they?

Reply:
for me the "topic" is "setting up the environment", which includes the dependencies and cron jobs